linkerd/service-profiles/src/http/mod.rs

```diff
@@ -67,7 +65,7 @@ impl Route {
         let labels = {
             let mut pairs = label_iter.collect::<Vec<_>>();
             pairs.sort_by(|(k0, _), (k1, _)| k0.cmp(k1));
-            Labels(Arc::new(IndexMap::from_iter(pairs)))
+            Labels(Arc::new(pairs.into_iter().collect()))
         };
```

Comment:
because we're now using `BTreeMap`, which is inherently sorted, we can skip the intermediary `Vec` here --- this can just be 
```rust
let labels = Arc::new(label_iter.collect());
```
(github won't let me leave a suggestion here because the diff contains deleted lines)